telegram/ext/_application.py

```diff
@@ -0,0 +1,1212 @@
+          :attr:`persistence` is set.
+
+        Note:
+            This does *not* start fetching updates from Telegram. You need to either start
+            :attr:`updater` manually or use one of :meth:`run_polling` or :meth:`run_webhook`.
+
+        Args:
+            ready (:obj:`asyncio.Event`, optional): If specified, the event will be set once the
+                application is ready.
+
+        Raises:
+            :exc:`RuntimeError`: If the application is already running or was not initialized.
+        """
+        if self.running:
+            raise RuntimeError('This Application is already running!')
+        if not self._initialized:
+            raise RuntimeError('This Application is not initialized!')
+
+        self._running = True
+        self.__update_persistence_event.clear()
+
+        try:
+            if self.persistence:
+                self.__update_persistence_task = asyncio.create_task(
+                    self._persistence_updater()
+                    # TODO: Add this once we drop py3.7
+                    # name=f'Application:{self.bot.id}:persistence_updater'
+                )
+                _logger.debug('Loop for updating persistence started')
+
+            if self.job_queue:
+                self.job_queue.start()
+                _logger.debug('JobQueue started')
+
+            self.__update_fetcher_task = asyncio.create_task(
+                self._update_fetcher(),
+                # TODO: Add this once we drop py3.7
+                # name=f'Application:{self.bot.id}:update_fetcher'
+            )
+            _logger.info('Application started')
+
+            if ready is not None:
+                ready.set()
+        except Exception as exc:
+            self._running = False
+            raise exc
+
+    async def stop(self) -> None:
+        """Stops the process after processing any pending updates or tasks created by
+        :meth:`create_task`. Also stops :attr:`job_queue`, if set.
+        Finally, calls :meth:`update_persistence` and :meth:`BasePersistence.flush` on
+        :attr:`persistence`, if set.
+
+        Warning:
+            Once this method is called, no more updates will be fetched from :attr:`update_queue`,
+            even if it's not empty.
+
+        Note:
+            This does *not* stop :attr:`updater`. You need to either manually call
+            :meth:`telegram.ext.Updater.stop` or use one of :meth:`run_polling` or
+            :meth:`run_webhook`.
+
+        Raises:
+            :exc:`RuntimeError`: If the application is not running.
+        """
+        if not self.running:
+            raise RuntimeError('This Application is not running!')
+
+        self._running = False
+        _logger.info('Application is stopping. This might take a moment.')
+
+        if self.updater and self.updater.running:
+            _logger.debug('Waiting for updater to stop fetching updates')
+            await self.updater.stop()
```

Comment:
The `Note:` says that this does not call `updater.stop()` but it does?

Reply:
right … don't know why I did that. moving to `run_*`